### PR TITLE
Added Auto

### PR DESCRIPTION
**Pull request by jeffaburke**
Use this differential drive for Francois Map

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -64,11 +67,12 @@
...
-        nameEntry.setPersistent();
-        nameEntry.setDefaultString("Unknown");
-        Shuffleboard.getTab("RobotData").addString("RobotName", () -> nameEntry.getString("Unknown"));
-
         autoChooser.setDefaultOption("Nothing", new InstantCommand());
+        autoChooser.addOption("Pass the Line", passLine());
+
+        Shuffleboard.getTab("Shuffleboard").add("Autonomous", autoChooser);
+
+        Shuffleboard.getTab("Shuffleboard").add("Autonomous", autoChooser);
```
Comment (bot190) [inline src/main/java/frc/robot/Robot.java]: Is this mean to appear twice?

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return new DifferentialDriveMap() {
+
+            @Override
+            public EncodedSpeedController getRight() {
+                CANSparkMax leader = new CANSparkMax(20, MotorType.kBrushless);
+                CANSparkMax follower = new CANSparkMax(21, MotorType.kBrushless);
+                CANEncoder leadEncoder = new CANEncoder(leader, EncoderType.kQuadrature, 42);
+                follower.follow(leader);
+
+                final double distancePerPulse = (1.0 / 46.0) * (1.0 / 12.27) * (6.0 * Math.PI);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java]: This should be defined up a level so it can be shared between the left and right.

**Review by bot190 — COMMENTED** [review]
I'm not sure why FrancoisMap looks to be new here, just a few comments otherwise.

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (jeffaburke) [reply src/main/java/frc/robot/Robot.java]: Matt added that

File: src/main/java/frc/robot/maps/FrancoisMap.java (hunk 2 above)
Comment (jeffaburke) [reply src/main/java/frc/robot/maps/FrancoisMap.java]: Finished

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 3]
```diff
@@ -18,14 +18,20 @@ public DifferentialDriveMap getDriveMap() {
             public EncodedSpeedController getRight() {
                 final SendableSpeedController rightGroup = SendableSpeedController.group(new WPI_TalonSRX(2),
                         new WPI_TalonSRX(3));
-                return EncodedSpeedController.join(rightGroup, new MockEncoder());
+                final WEncoder encoder = new WEncoder(1, 0);
+                final double distancePerPulse = (1.0 / 360.0) * (4.0 * Math.PI);
```
Comment (msoucy) [inline src/main/java/frc/robot/maps/TempestMap.java]: Missed a spot

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 4]
```diff
@@ -3,10 +3,13 @@
...
 
 import com.chopshop166.chopshoplib.maps.DifferentialDriveMap;
-import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
+import com.chopshop166.chopshoplib.outputs.EncodedSpeedController;
 
+import edu.wpi.first.wpilibj.GyroBase;
 import edu.wpi.first.wpilibj.drive.DifferentialDrive;
 import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: Clean up imports

**Review by msoucy — CHANGES_REQUESTED** [review]

**Review by AustinFortunato — APPROVED** [review]

**Review by AustinFortunato — DISMISSED** [review]

File: src/main/java/frc/robot/maps/TempestMap.java (hunk 3 above)
Comment (jeffaburke) [reply src/main/java/frc/robot/maps/TempestMap.java]: Fixed

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 4 above)
Comment (jeffaburke) [reply src/main/java/frc/robot/subsystems/Drive.java]: Fixed

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 5]
```diff
@@ -59,7 +63,30 @@ public CommandBase drive(DoubleSupplier forward, DoubleSupplier turn) {
...
\ No newline at end of file
+    public CommandBase turnDegrees(double degrees, double speed) {
+        return new FunctionalCommand(() -> {
+            gyro.reset();
+        }, () -> {
+            driveTrain.arcadeDrive(0, speed);
+        }, (interrupted) -> {
+            driveTrain.stopMotor();
+        }, () -> {
+            return gyro.getAngle() >= degrees;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: How will this behave if `degrees` is negative?

Comment (jeffaburke) [reply src/main/java/frc/robot/subsystems/Drive.java]: Finished

**Review by msoucy — APPROVED** [review]

**Pull request merged**